### PR TITLE
[BEE-49946] Add data migration sizing script

### PR DESCRIPTION
**Pull request by jsmith-cb**
Jira: [BEE-49946](https://cloudbees.atlassian.net/browse/BEE-49946)

[BEE-49946]: https://cloudbees.atlassian.net/browse/BEE-49946?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by chpock — APPROVED** [review]

File: cloudbees-cd/es-to-os-migration/README.md [hunk 1]
```diff
@@ -1 +1,98 @@
...
+
+To help roughly estimate the size of legacy CloudBees Analytics indexes data that will be migrated to your updated CloudBees Analytics server,  CloudBees has provided the [reporting-data-reindex.pl](reporting-data-reindex.pl) utility.
+
+> **IMPORTANT**  
+> As described in the [known issues of the data migration](https://docs.cloudbees.com/docs/cloudbees-cd/latest/troubleshooting/data-migration-es-to-os#data-migration-known-issues), the automated processes CloudBees has supplied to automate data migration have a timeout of `180 minutes` per index. This is roughly equivalent to `32,000,000` documents, which may vary based on the complexity of individual documents (number of fields, etc.).   
+> 
+> If your indexes contain vastly more than 32,000,000 documents, or you reach the timeout limit for specific indexes, it may be necessary to split the specific index into a set of smaller indexes. For more information on this process, refer to the [Elasticsearch Split index API](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/indices-split-index.html) documentation.
```
Comment (urvashisingh) [inline cloudbees-cd/es-to-os-migration/README.md:13]: Unfortunately using the spit index API will not do what we need here. The spit index API spits the documents in the index into more shards by increasing the number of primary shards. But the total number of documents in the index stay the same. The fix we need is to split the documents into multiple indexes...let me think about this and get back to you if we can document some instructions for users on how to do this. For now, we can remove the line on split index API.

File: cloudbees-cd/es-to-os-migration/README.md [hunk 2]
```diff
@@ -1 +1,98 @@
...
+$ /opt/cloudbees/sda/bin/cb-perl ./reporting-data-reindex.pl --showStatistics=1 --sourceUrl=https://<hostname>:9200 --sourceAuthUser=reportuser --sourceAuthPassword=<password>
+```
+
+Where:
+
+* `--showStatistics=1`: The required parameter, which selects the mode of displaying index statistics.
+
+* `--sourceUrl=https://<hostname>:9200`: Specifies URL for the legacy CloudBees Analytics server.
+
+* `--sourceAuthUser=reportuser`: Specifies the username to authenticate to the legacy CloudBees Analytics server. But default, it is `reportuser`.
```
Comment (urvashisingh) [inline cloudbees-cd/es-to-os-migration/README.md:34]:
```suggestion
* `--sourceAuthUser=reportuser`: Specifies the username to authenticate to the legacy CloudBees Analytics server. By default, it is `reportuser`.
```

Comment (jsmith-cb) [reply cloudbees-cd/es-to-os-migration/README.md]: Updated with [3474237](https://github.com/cloudbees/cloudbees-examples/pull/81/commits/3474237df126100a21f873735287d5f8cbfaefd5).

**Review by urvashisingh — APPROVED** [review]

File: cloudbees-cd/es-to-os-migration/README.md [hunk 3]
```diff
@@ -1 +1,97 @@
-Placeholder path for ES to OS README
+# Estimate data size of legacy CloudBees Analytics indexes
+
+In CloudBees CD/RO v2024.06.0, the CloudBees Analytics server was upgraded from Elasticsearch to OpenSearch. As part of the upgrade process from v2024.03.0 or earlier to v2024.06.0 or later, you must migrate your CloudBees Analytics to OpenSearch, as described on:
```
Comment (MargoBGH) [inline cloudbees-cd/es-to-os-migration/README.md:3]:
```suggestion
In CloudBees CD/RO v2024.06.0, the CloudBees Analytics server was upgraded from Elasticsearch to OpenSearch. As part of the upgrade process from v2024.03.0 or earlier to v2024.06.0 or later, you must migrate your CloudBees Analytics to OpenSearch, as described in:
```

File: cloudbees-cd/es-to-os-migration/README.md [hunk 4]
```diff
@@ -1 +1,97 @@
...
+
+* [Traditional migrations](https://docs.cloudbees.com/docs/cloudbees-cd/latest/troubleshooting/trad-upgrade-to-os)
+* [Kubernetes migrations](https://docs.cloudbees.com/docs/cloudbees-cd/latest/troubleshooting/k8s-upgrade-to-os)
+
+To help roughly estimate the size of legacy CloudBees Analytics indexes data that will be migrated to your updated CloudBees Analytics server,  CloudBees has provided the [reporting-data-reindex.pl](reporting-data-reindex.pl) utility.
```
Comment (MargoBGH) [inline cloudbees-cd/es-to-os-migration/README.md:8]:
```suggestion
CloudBees has provided the [reporting-data-reindex.pl](reporting-data-reindex.pl) utility to help you roughly estimate the size of legacy CloudBees Analytics indexes data that will be migrated to your updated CloudBees Analytics server.
```

File: cloudbees-cd/es-to-os-migration/README.md [hunk 5]
```diff
@@ -1 +1,97 @@
...
+
+```sh
+$ /opt/cloudbees/sda/bin/cb-perl ./reporting-data-reindex.pl --showStatistics=1 --sourceUrl=https://<hostname>:9200 --sourceAuthUser=reportuser --sourceAuthPassword=<password>
+```
+
+Where:
+
+* `--showStatistics=1`: The required parameter, which selects the mode of displaying index statistics.
+
+* `--sourceUrl=https://<hostname>:9200`: Specifies URL for the legacy CloudBees Analytics server.
```
Comment (MargoBGH) [inline cloudbees-cd/es-to-os-migration/README.md:31]:
```suggestion
* `--sourceUrl=https://<hostname>:9200`: Specifies the legacy CloudBees Analytics server URL.
```

File: cloudbees-cd/es-to-os-migration/README.md [hunk 6]
```diff
@@ -1 +1,97 @@
...
+
+* `--sourceAuthUser=reportuser`: Specifies the username to authenticate to the legacy CloudBees Analytics server. By default, it is `reportuser`.
```
Comment (MargoBGH) [inline cloudbees-cd/es-to-os-migration/README.md:33]:
```suggestion
* `--sourceAuthUser=reportuser`: This setting specifies the username to authenticate to the legacy CloudBees Analytics server. By default, it is `reportuser`.
```

**Review by MargoBGH — APPROVED** [review]
lgtm!

File: cloudbees-cd/es-to-os-migration/README.md [hunk 7]
```diff
@@ -1 +1,97 @@
...
+
+The format for launching the utility is as follows:
+
+```sh
+$ /opt/cloudbees/sda/bin/cb-perl ./reporting-data-reindex.pl --showStatistics=1 --sourceUrl=https://<hostname>:9200 --sourceAuthUser=reportuser --sourceAuthPassword=<password>
+```
+
+Where:
+
+* `--showStatistics=1`: The required parameter, which selects the mode of displaying index statistics.
```
Comment (jsmith-cb) [inline cloudbees-cd/es-to-os-migration/README.md:29]:
```suggestion
* `--showStatistics=1`: *_Required_*. Specifies to display the index statistics.
```

File: cloudbees-cd/es-to-os-migration/README.md [hunk 8]
```diff
@@ -1 +1,97 @@
...
+# Estimate data size of legacy CloudBees Analytics indexes
+
+In CloudBees CD/RO v2024.06.0, the CloudBees Analytics server was upgraded from Elasticsearch to OpenSearch. As part of the upgrade process from v2024.03.0 or earlier to v2024.06.0 or later, you must migrate your CloudBees Analytics to OpenSearch, as described in:
+
+* [Traditional migrations](https://docs.cloudbees.com/docs/cloudbees-cd/latest/troubleshooting/trad-upgrade-to-os)
+* [Kubernetes migrations](https://docs.cloudbees.com/docs/cloudbees-cd/latest/troubleshooting/k8s-upgrade-to-os)
+
+CloudBees has provided the [reporting-data-reindex.pl](reporting-data-reindex.pl) utility to help you roughly estimate the size of legacy CloudBees Analytics indexes data that will be migrated to your updated CloudBees Analytics server.
+
+> **IMPORTANT**
```
Comment (jsmith-cb) [inline cloudbees-cd/es-to-os-migration/README.md:10]:
```suggestion
> [!IMPORTANT]
```

File: cloudbees-cd/es-to-os-migration/README.md [hunk 9]
```diff
@@ -1 +1,97 @@
...
+
+## Dependencies
+This utility requires `cb-perl` to run, and is executable in any environment with a CloudBees CD/RO server or agent installation.
+
+## Command Format
+
+The format for launching the utility is as follows:
+
+```sh
+$ /opt/cloudbees/sda/bin/cb-perl ./reporting-data-reindex.pl --showStatistics=1 --sourceUrl=https://<hostname>:9200 --sourceAuthUser=reportuser --sourceAuthPassword=<password>
```
Comment (urvashisingh) [inline cloudbees-cd/es-to-os-migration/README.md:24]:
```suggestion
$ <cdro_install_dir>/bin/cb-perl ./reporting-data-reindex.pl --showStatistics=1 --sourceUrl=https://<hostname>:9200 --sourceAuthUser=reportuser --sourceAuthPassword=<password>
```

File: cloudbees-cd/es-to-os-migration/README.md [hunk 10]
```diff
@@ -1 +1,97 @@
...
+```
+
+Where:
+
```
Comment (urvashisingh) [inline cloudbees-cd/es-to-os-migration/README.md:28]:
```suggestion
* <cdro_install_dir>: Directory where CD/RO server or agent is installed.
```

**Review by urvashisingh — APPROVED** [review]

**Pull request merged**